crates/core/src/factories/core_fetch.rs

```diff
@@ -295,12 +296,19 @@ impl CoreFetch {
         outgoing_request_tx: Sender<OutgoingRequest>,
         outgoing_request_rx: Arc<tokio::sync::Mutex<Receiver<OutgoingRequest>>>,
         space_id: SpaceId,
-        transport: DynTransport,
+        transport: Weak<dyn Transport>,
         re_insert_outgoing_request_delay: u32,
     ) {
         while let Some((op_id, peer_url)) =
             outgoing_request_rx.lock().await.recv().await
         {
+            let Some(transport) = transport.upgrade() else {
```

Comment:
wuuuuuut is this syntax ?!?/*&#* :exploding_head:

Reply:
Cool right :) It lets you flatten out the let/if/let/else/return that is so often a useful pattern but ends up with nested code. I saw it in a diff on some other project when trying to track down a bug and decided it was worth remembering :D